transpiler/src/ast/core/BitloopsVisitor/helpers/aggregateDeclarationVisitor.ts

```diff
@@ -1,18 +1,24 @@
 import BitloopsParser from '../../../../parser/core/grammar/BitloopsParser.js';
-import { TEntityValues } from '../../../../types.js';
+import { RootEntityDeclarationNodeBuilder } from '../../intermediate-ast/builders/RootEntity/RootEntityDeclarationBuilder.js';
+import { RootEntityIdentifierNodeBuilder } from '../../intermediate-ast/builders/RootEntity/RootEntityIdentifierBuilder.js';
 import BitloopsVisitor from '../BitloopsVisitor.js';
+import { produceMetadata } from '../metadata.js';
 
 export const aggregateDeclarationVisitor = (
   thisVisitor: BitloopsVisitor,
-  ctx: BitloopsParser.EntityDeclarationContext,
-): { RootEntities: { [identifier: string]: TEntityValues } } => {
+  ctx: BitloopsParser.AggregateDeclarationContext,
+): void => {
+  const metadata = produceMetadata(ctx, thisVisitor);
+
   const valueObjectIdentifier = ctx.entityIdentifier().getText();
```

Comment:
Replace this with `thisVisitor.visit(ctx. entityIdentifier());` . There is a fix that it will be merged to the refactoring-builder branch. This is for the identifier to take the right metadata.